test/sql-tap/alter2.test.lua

```diff
@@ -224,15 +224,20 @@ test:do_catchsql_test(
         CREATE TABLE t1 (id INT PRIMARY KEY);
         ALTER TABLE t1 ADD CONSTRAINT ck CHECK(id > 0);
         INSERT INTO t1 VALUES (-1);
-    ]], { 1, "Check constraint failed 'CK': id > 0" })
+    ]], { 1, "Check constraint 'CK' failed for tuple" })
 
 -- Make sure that one can't create constraint with the same name twice.
 --
 test:do_catchsql_test(
     "alter2-6.2",
     [[
         ALTER TABLE t1 ADD CONSTRAINT ck CHECK(id > 0);
-    ]], { 1, "CHECK constraint 'CK' already exists in space 'T1'" })
+    ]], { 1, 'Duplicate key exists in unique index "name" in space "_func" '..
+         'with old tuple - [66, 1, "check_T1_CK", 0, "SQL_EXPR", "id > 0", '..
+         '"function", [], "any", "none", "none", true, false, true, ["LUA"], '..
+         '{}, "", "", ""] and new tuple - [67, 1, "check_T1_CK", 0, '..
+         '"SQL_EXPR", "id > 0", "function", [], "any", "none", "none", true, '..
+         'false, true, ["LUA"], {}, "", "", ""]'})
```

Comment:
I remember there was an explicit intention to keep the error messages human readable even when update system spaces under the hood. This is why we have `vdbe_emit_halt_with_presence_test()` IIRC. We should probably keep the nice error message.

Reply:
Fixed.